docs/shared/recipes/running-tasks/customizing-inputs.md

```diff
@@ -44,136 +48,282 @@ To understand some of the examples below, let's imagine the following simple wor
+
+These sets of inputs can then be referenced and reused in the `inputs` array of targets. You can think of these as variables which can be used in the `inputs` array.
+
+Named Inputs can be defined for the entire workspace in `nx.json` or for a specific project in `project.json`.
+
+Once defined for the workspace, any tasks for all projects in the workspace can use the named inputs in the `inputs` array.
+
+{% tabs %}
+{% tab label="Default Named Inputs" %}
+
+By default, Nx Workspaces are generated with the following named inputs:
+
+```jsonc {% fileName="nx.json" highlightLines=["2-6"] %}
 {
-  "name": "some-project",
-  "dependencies": {},
-  "devDependencies": {},
-  ...
-  "nx": {
-    "targets": {
-      "build": {
-        "inputs": ["..."],
-        ...
-      }
-      ...
-    }
+  "namedInputs": {
+    "default": ["{projectRoot}/**/*", "sharedGlobals"], // Default Inputs
+    "production": ["default", "!{projectRoot}/some-dev-only-files"], // Production Inputs
+    "sharedGlobals": [] // Shared Global Inputs
+  },
+  "targetDefaults": {
+    "inputs": ["default", "^default"]
   }
 }
 ```
 
 {% /tab %}
+{% tab label="Workspace (nx.json)" %}
 
-## Include all project files and dependencies in cache hash
+The following example shows how to define named inputs for the entire workspace in `nx.json`.
 
-The following definition includes all files of the project itself as well as all of its dependencies in the cache hash, hence telling Nx to invalidate the cache whenever
+```jsonc {% fileName="nx.json" highlightLines=["3-6"] %}
+{
+  "namedInputs": {
+    "default": ["{projectRoot}/**/*"] // Default Inputs
+  }
+}
+```
 
-- any file of the project itself changes
-- any file of any dependency changes
+{% /tab %}
+{% tab label="Project Level (project.json)" %}
 
-```jsonc {% fileName="nx.json" %}
+```jsonc {% fileName="nx.json" highlightLines=["2-6"] %}
 {
   "namedInputs": {
-    "default": ["{projectRoot}/**/*"]
+    "default": ["{projectRoot}/**/*", "sharedGlobals"], // Default Inputs
+    "production": ["default", "!{projectRoot}/some-dev-only-files"], // Production Inputs
+    "sharedGlobals": [] // Shared Global Inputs
   },
   "targetDefaults": {
-    "build": {
-      "inputs": ["default", "^default"]
-    }
+    "inputs": ["default", "^default"]
   }
 }
 ```
 
-This definition is the default behavior of Nx, even if you don't specify any `inputs` at all. This is a rather cautious approach. It will always give you correct output, but it might re-run a task in some cases where the cache could have been used instead.
+{% /tab %}
+{% tab %}
+Naming a set of inputs with the same name as a set of inputs for the workspace will override the meaning of that set for the project.
+{% /tab %}
+{% /tabs %}
 
-Note, Nx uses the [minimatch](https://github.com/isaacs/minimatch) library to process glob patterns.
+### Default Inputs
```

Comment:
Default, Shared Global and Production inputs should be under the "Default Named inputs section" so it is clear they are only default names and can be overriden